Sources/NIO/ChannelHandlers.swift

```diff
@@ -17,6 +17,80 @@
 
 import struct Dispatch.DispatchTime
 
+
+/// `ChannelHandler` which implements some sort of backoff for a `ServerChannel` once `accept` will produce an `IOError`. This is useful as often these errors are recoverable in the sense that we can recover if we slow down the accept rate a bit.
+public final class AcceptBackoffHandler: ChannelDuplexHandler {
+    public typealias InboundIn = Channel
+    public typealias OutboundIn = Channel
+    
+    private var currentBackoffTime: Int?
+    private let backoffProvider: (IOError) -> TimeAmount?
+    private var scheduledRead: Scheduled<Void>?
+    
+    /// Create a new instance
+    ///
+    /// - parameters:
+    ///     - backoffProvider: returns a `TimeAmount` to backoff more `reads` (a.k.a accepts).
+    public init(backoffProvider: @escaping (IOError) -> TimeAmount?) {
+        self.backoffProvider = backoffProvider
+    }
+    
+    public func read(ctx: ChannelHandlerContext) {
+        // If we already have a read scheduled there is no need to schedule another one.
+        if scheduledRead == nil {
+            if let backoffTime = currentBackoffTime {
+                let now = Int(DispatchTime.now().uptimeNanoseconds)
+                if now >= backoffTime {
+                    // The backoff already expired, just do a read.
+                    self.doRead(ctx)
+                } else {
+                    // Schedule the read to be executed after the backoff time elapsed.
+                    scheduleRead(TimeAmount.nanoseconds(backoffTime - now), ctx)
+                }
+            }
+        }
+    }
+    
+    public func errorCaught(ctx: ChannelHandlerContext, error: Error) {
+        if let ioError = error as? IOError {
+            if let amount = backoffProvider(ioError) {
+                currentBackoffTime = Int(DispatchTime.now().uptimeNanoseconds) + amount.nanoseconds
+                if let scheduled = scheduledRead {
+                    scheduled.cancel()
+                    scheduleRead(TimeAmount.nanoseconds(currentBackoffTime!), ctx)
```

Comment:
here you schedule a read with a time in nanoseconds _absolute_ (including `now`). So something can't be right here.

Reply:
`TimeInterval` is supposed to be relative to now not an absolute amount of time.